instrumentation/opentelemetry/github.com/jackc/hyperpgx/pgx.go

```diff
@@ -0,0 +1,149 @@
+package hyperpgx
+
+import (
+	"context"
+	"database/sql/driver"
+
+	"github.com/hypertrace/goagent/instrumentation/opentelemetry"
+	"github.com/hypertrace/goagent/sdk"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgtype/pgxtype"
+	"github.com/jackc/pgx/v4"
+)
+
+var _ PGXConn = (*pgx.Conn)(nil)
+
+// PGXConn contains all public methods included by *pgx.Conn as an attempt to make the instrumentation transparent
+// for the user.
+type PGXConn interface {
+	pgxtype.Querier
+	driver.Pinger
+
+	// QueryFunc executes sql with args. For each row returned by the query the values will scanned into the elements of
+	// scans and f will be called. If any row fails to scan or f returns an error the query will be aborted and the error
+	// will be returned.
+	QueryFunc(ctx context.Context, sql string, args []interface{}, scans []interface{}, f func(pgx.QueryFuncRow) error) (pgconn.CommandTag, error)
+
+	// SendBatch sends all queued queries to the server at once. All queries are run in an implicit transaction unless
+	// explicit transaction control statements are executed. The returned BatchResults must be closed before the connection
+	// is used again.
+	SendBatch(ctx context.Context, b *pgx.Batch) pgx.BatchResults
+
+	// Close closes a connection. It is safe to call Close on a already closed
+	// connection.
+	Close(ctx context.Context) error
+}
+
+var _ PGXConn = (*wrappedConn)(nil)
+
+type wrappedConn struct {
+	delegate  *pgx.Conn
+	connAttrs map[string]string
+}
+
+var _ pgx.Row = (*wrappedRow)(nil)
+
+type wrappedRow struct {
+	delegate pgx.Row
+	span     sdk.Span
+}
+
+func (r *wrappedRow) Scan(dest ...interface{}) error {
+	err := r.delegate.Scan(dest...)
+	if err != nil {
+		r.span.SetError(err)
+	}
+
+	return err
+}
+
+func (w *wrappedConn) Query(ctx context.Context, query string, optionsAndArgs ...interface{}) (pgx.Rows, error) {
+	ctx, span, closer := opentelemetry.StartSpan(ctx, "db:query", &sdk.SpanOptions{Kind: sdk.Client})
+	defer closer()
+
+	for k, v := range w.connAttrs {
+		span.SetAttribute(k, v)
+	}
+	span.SetAttribute("db.statement", query)
+
+	rows, err := w.delegate.Query(ctx, query, optionsAndArgs...)
```

Comment:
is the response of no use to us?

Reply:
for sql, we don't need to capture the response.  we're only interested in the query.